src/test/regress/expected/union_pushdown.out

```diff
@@ -874,6 +876,123 @@ SELECT create_distributed_table('test_b','id');
 
 (1 row)
 
+-- try with composite types
+CREATE TYPE comp_type AS (
+    int_field_1 BIGINT,
+    int_field_2 BIGINT
+);
+CREATE TABLE range_dist_table_2 (dist_col comp_type);
+SELECT create_distributed_table('range_dist_table_2', 'dist_col', 'range');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+CALL public.create_range_partitioned_shards('range_dist_table_2',
+    '{"(10,24)","(10,58)",
+	  "(10,90)","(20,100)"}',
+	'{"(10,25)","(10,65)",
+	  "(10,99)","(20,100)"}');
+INSERT INTO range_dist_table_2 VALUES ((10, 24));
+INSERT INTO range_dist_table_2 VALUES ((10, 60));
+INSERT INTO range_dist_table_2 VALUES ((10, 91));
+INSERT INTO range_dist_table_2 VALUES ((20, 100));
+-- the following can be pushed down
+CREATE OR REPLACE VIEW v2 AS SELECT * from range_dist_table_2 UNION ALL SELECT * from range_dist_table_2;
+SELECT public.explain_has_distributed_subplan($$
+EXPLAIN
+SELECT COUNT(dist_col) FROM v2;
+$$);
+ explain_has_distributed_subplan
+---------------------------------------------------------------------
+ f
+(1 row)
+
+SELECT public.explain_has_distributed_subplan($$
+EXPLAIN
+SELECT COUNT(*) FROM v2;
+$$);
+ explain_has_distributed_subplan
+---------------------------------------------------------------------
+ f
+(1 row)
+
+DROP TABLE range_dist_table_2 cascade;
+NOTICE:  drop cascades to view v2
+-- these should be pushed down.
+SELECT public.explain_has_distributed_subplan($$
+explain SELECT * FROM users_table_part u1 WHERE (user_id) IN
+(
+SELECT user_id FROM users_table_part
+UNION
+SELECT user_id FROM users_table_part
+) ;
+$$);
+ explain_has_distributed_subplan
+---------------------------------------------------------------------
+ f
+(1 row)
+
+SELECT public.explain_has_distributed_subplan($$
+explain SELECT * FROM users_table_part u1 WHERE (user_id) IN
+(
+SELECT user_id FROM (SELECT * FROM users_table_part) as foo
```

Comment:
I guess we should add something to prevent pulling up the subquery to `foo`, such as `SELECT *, random() FROM ..`